Zotero/Scenes/AppCoordinator.swift

```diff
@@ -121,6 +122,16 @@ final class AppCoordinator: NSObject {
         self.process(connectionOptions: options, session: session)
     }
 
+    private func preprocess(connectionOptions: UIScene.ConnectionOptions?, session: UISceneSession?) {
+        guard let userActivity = connectionOptions?.userActivities.first ?? session?.stateRestorationActivity, let data = userActivity.restoredStateData else { return }
+        // If scene had state stored, check if defaults need to be updated first
+        DDLogInfo("AppCoordinator: Preprocessing restored state - \(data)")
+        Defaults.shared.selectedLibrary = data.libraryId
+        if let collectionId = data.collectionId {
+            Defaults.shared.selectedCollectionId = collectionId
+        }
+    }
+    
     private func process(connectionOptions: UIScene.ConnectionOptions, session: UISceneSession) {
```

Comment:
We load the `RestoredStateData` in new `preprocess(connectionOptions:session:)` function and then load them again a few lines later here. Maybe the `preprocess` function could return `RestoredStateData?` and they could just be passed to `process(restoredState data: RestoredStateData, urlContext: UIOpenURLContext)` after `self.show(viewController: viewController, in: window, animated: animated)` happens.

Reply:
Check https://github.com/zotero/zotero-ios/pull/713/commits/76c09809ca57bccf4a83c63cbde1c739161141d2